unittest/gunit/keyring/keyring-api-t.cc

```diff
@@ -364,82 +321,81 @@ TEST_F(Keyring_api_test, GeneratePBRotatePBFetchFirstVersionFetchLatestPB) {
 
   char *key_type = NULL;
   size_t key_len = 0;
-  void *key_ver1 = NULL;
+  void *key_ver0 = NULL;
 
   EXPECT_EQ(
-      mysql_key_fetch("percona_binlog:1", &key_type, NULL, &key_ver1, &key_len),
+      mysql_key_fetch("percona_binlog:0", &key_type, NULL, &key_ver0, &key_len),
       0);
   EXPECT_STREQ("AES", key_type);
   EXPECT_EQ(key_len, static_cast<size_t>(16));
   my_free(key_type);
   key_type = NULL;
 
-  void *key_ver2 = NULL;
+  void *key_ver1 = NULL;
 
   EXPECT_EQ(
-      mysql_key_fetch("percona_binlog", &key_type, NULL, &key_ver2, &key_len),
+      mysql_key_fetch("percona_binlog", &key_type, NULL, &key_ver1, &key_len),
       0);
   EXPECT_STREQ("AES", key_type);
   EXPECT_EQ(key_len, static_cast<size_t>(18));
-  ASSERT_TRUE(memcmp(reinterpret_cast<char *>(key_ver2), "2:", 2) == 0);
+  ASSERT_TRUE(memcmp(reinterpret_cast<char *>(key_ver1), "1:", 2) == 0);
   my_free(key_type);
   key_type = NULL;
 
   // make sure that rotated key is different than the original one
-  // + 2 to skip key version of retrieved latest percona_binlog key
-  ASSERT_TRUE(memcmp(reinterpret_cast<char *>(key_ver1),
-                     reinterpret_cast<char *>(key_ver2) + 2, 16) != 0);
+  ASSERT_TRUE(memcmp(reinterpret_cast<char *>(key_ver0) + 2,
```

Comment:
@dutow, please consider changing this line to 

ASSERT_TRUE(memcmp(reinterpret_cast<char *>(key_ver0),

ASAN complains here

`==342140==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x6040000126c0 at pc 0x7f33f46abf50 bp 0x7ffd4a052b90 sp 0x7ffd4a052340
READ of size 16 at 0x6040000126c0 thread T0
    #0 0x7f33f46abf4f in MemcmpInterceptorCommon(void*, int (*)(void const*, void const*, unsigned long), void const*, void const*, unsigned long) (/lib64/libasan.so.8+0xabf4f)
    #1 0x7f33f46ac5e8 in memcmp (/lib64/libasan.so.8+0xac5e8)
    #2 0x6335ee in keyring__api_unittest::Keyring_api_test_GeneratePBRotatePBFetchFirstVersionFetchLatestPB_Test::TestBody() unittest/gunit/keyring/keyring-api-t.cc:346`